ontopy/utils.py

```diff
@@ -391,12 +391,22 @@ def write_catalog(
             (URIs).  It has the same format as the dict returned by
             read_catalog().
         output: name of catalog file.
-        dir: directory path to the catalog file.  Only used if `output`
+        directory: directory path to the catalog file.  Only used if `output`
             is a relative path.
+        relative_paths: whether to write aboslute paths or paths relative to
+            directory for local paths inside the catalog file.
         append: whether to append to a possible existing catalog file.
             If false, an existing file will be overwritten.
     """
-    filename = (Path(dir) / output).resolve()
+    web_protocol = "http://", "https://", "ftp://"
+    if relative_paths:
+        for key, item in mappings.items():
+            mappings[key] = (
+                item
+                if item.startswith(web_protocol)
+                else os.path.relpath(item, Path(directory).resolve())
```

Comment:
```suggestion
            if not item.startswith(web_protocol):
                mappings[key] = os.path.relpath(item, Path(directory).resolve())
```

Reply:
Added by means of script, for prior testing.